cql3/expr/expression.cc

```diff
@@ -931,6 +931,61 @@ std::ostream& operator<<(std::ostream& s, oper_t op) {
     __builtin_unreachable();
 }
 
+extern std::vector<expression> extract_column_restrictions(const expression& expr, const column_definition& column) {
+    struct visitor {
+        std::vector<expression> restrictions;
+        const column_definition& column;
+        const binary_operator* current_binary_operator;
+
+        void operator()(bool b) {
+            restrictions.emplace_back(b);
+        }
+
+        void operator()(const conjunction& conj) {
+            for (const expression& child : conj.children) {
+                std::visit(*this, child);
+            }
+        }
+
+        void operator()(const binary_operator& oper) {
+            if (current_binary_operator != nullptr) {
+                on_internal_error(expr_logger, "extract_column_restrictions: nested binary operators are not supported");
+            }
+
+            current_binary_operator = &oper;
+            std::visit(*this, *oper.lhs);
+            current_binary_operator = nullptr;
+        }
+
+        void operator()(const column_value& cv) {
+            if (*cv.col == column && current_binary_operator != nullptr) {
+                restrictions.emplace_back(*current_binary_operator);
+            }
```

Comment:
The code is correct as is, but not forward-looking.

In SQL (but no CQL) you can write

```sql
SELECT *
FROM tab
WHERE bool_column
```

so you have a column_value which is not in a binary operator, and is a restriction on that column.

We can enable it when the time comes.